### PR TITLE
Removes deprecated ErrNilNextConsumer and solves type error for typeStr

### PR DESCRIPTION
**Pull request by tarsojabbes**
This PR corrects type error for typeStr on Custom Receiver documentation, as well as removes ErrNilNextConsumer checks.

As of version 0.97.0 for dependencies when creating a custom receiver, using typeStr directly as a string was throwing an error of `cannot use typeStr (variable of type string) as component.Type value in argument to receiver.NewFactory ` which should now be done using `component.MustNewType("tailtracer")`, in a var declaration.

Also, as in [Issue 9322](https://github.com/open-telemetry/opentelemetry-collector/issues/9322) and later on accept as [PR 9779](https://github.com/open-telemetry/opentelemetry-collector/pull/9779) and [PR 31793](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31793),` component.ErrNilNextConsumer` is not a valid error type.

### CODE REVIEW

**Review by theletterf — APPROVED** [review]
@open-telemetry/collector-contrib-approvers PTAL

File: content/en/docs/collector/building/receiver.md [hunk 1]
```diff
@@ -397,8 +397,8 @@ value for it so it can be used as part of the default settings.
 Go ahead and add the following code to your `factory.go` file:
 
 ```go
-const (
-	typeStr         = "tailtracer"
+var (
+	typeStr         = component.MustNewType("tailtracer")
 	defaultInterval = 1 * time.Minute
 )
```
Comment (mx-psi) [inline content/en/docs/collector/building/receiver.md:403]: nit: We could leave `defaultInterval` as `const`

**Comment by mjnowen** [comment]
+1 this PR LGTM

File: content/en/docs/collector/building/receiver.md (hunk 1 above)
Comment (tarsojabbes) [reply content/en/docs/collector/building/receiver.md]: Fixed! Can you take a second look?

**Review by mx-psi — APPROVED** [review]

**Review by bryan-aguilar — APPROVED** [review]

**Review by cartermp — APPROVED** [review]

**Pull request merged**